packages/runner/src/builtins/map.ts

```diff
@@ -97,6 +97,9 @@ export function map(
       throw new Error("map currently only supports arrays");
     }
 
+    const newArrayValue = [
+      ...resultWithLog.getRaw()!.slice(0, initializedUpTo),
```

Comment:
you have .getRaw() here, but .set() below. This only currently works because we're allowing sigil links to be set, but I have a ticket to fix that (it's a security violation if non-privileged code can set arbitrary links)